src/librustc/infer/error_reporting/need_type_info.rs

```diff
@@ -166,4 +159,27 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
 
         err
     }
+
+    pub fn need_type_info_err_in_generator(&self,
```

Comment:
Nit: Can we follow the rustfmt conventions here?

i.e., move `&self` etc to the next line

```rust
fn need_type_info_err_in_generator(
    &self,
    span: Span,
    ty: Ty<'tcx>,
) -> DiagnosticBuilder<'gcx> {
}
```

Reply:
Agreed, looks much nicer that way.